modules/openthread/platform/diag.c

```diff
@@ -104,8 +104,8 @@ bool otPlatDiagModeGet(void)
 
 void otPlatDiagChannelSet(uint8_t aChannel)
 {
-	ARG_UNUSED(aChannel);
 	sChannel = aChannel;
+	platformRadioChannelSet(aChannel);
```

Comment:
This doesn't seem related to state machine fixing? IMO the channel setting fix merits a separate commit.